### ISSUE TITLE
[fr] add support for ".opus" files

### ISSUE THREAD

**Issue by throwaway242685** [FossifyOrg/Music-Player#2]
**Checklist**
 - [x] I made sure that there are no existing issues - open or closed - to which I could contribute my information.
 - [x]  I have read the FAQ and my problem isn't listed.
 - [x]  I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
 - [x]  This issue contains only one feature request.
 - [x]  I have read and understood the [contribution guidelines.](https://github.com/FossifyOrg/General-Discussion?tab=readme-ov-file#faq)

**Is your feature request related to a problem? Please describe.**

this app doesn't seem to be able to detect `.opus` files, and they can't be played normally from the app.

all the songs that I've downloaded from NewPipe in `.opus` format can't be played here.

**Describe the solution you'd like**

make the app compatible with the `.opus` format.

**Describe alternatives you've considered**

_No response_

**Additional context**

_No response_

**Comment by Aga-C** [comment]
What phone and OS version are you using? I've tested it on my phone with Android 13 and it works.

Also, can you check the Voice Recorder and test if you can record in Opus format there? It might be a related problem.

**Comment by useless642** [comment]
@Aga-C you can't find these `.opus` files from SMP itself, it just doesn't detect them.

can you test if you can find the song from above from SMP itself? because it doesn't show up for me.

**Comment by Aga-C** [comment]
Yes, it detects and it's fully working. That's why I'm asking for your device and OS because it seems to be a problem specific for your device.

**Comment by Aga-C** [comment]
> can you test if you can find the song from above from the music player itself? because it doesn't show up for me.

Yes, it detects and it's fully working. Probably it's the same problem as in Voice Recorder which supports Opus from Android 10+.

Thanks!

**Comment by useless642** [comment]
> Also, can you check Voice Recorder and test if you can record in Opus format there?

about that... I don't even have that format as an option, lol.

![Screenshot_2023-12-23_19-46-58.png](https://github.com/FossifyOrg/Music-Player/assets/85243826/91f1bedc-c3cb-4e48-9243-cca1745e4e0c)

(and yes, the app is up to date)

**Comment by ulu** [comment]
Your problem is Android 9. 

According to [wikipedia](https://en.wikipedia.org/wiki/Opus_(audio_format)#Operating_system_support):

> Google added native support for Opus audio playback in Android 5.0. However, it was limited to Opus audio encapsulated in Matroska and WebM containers, such as .mkv, .mka and .webm files. Android 7.0 introduced support for Opus audio encapsulated in Ogg containers. Android 10 finally added native support for .opus extensions.

**Comment by firubat** [comment]
> this music player doesn't seem to be able to detect .opus files, and they can't be played normally from the app.

I'm having this same issue, on Fairphone 4 with Android 13.
The files can be played on VLC so I don't think it's an OS issue. Should I open a separate issueor can this beconverted to a bug? [It probably should be closed as a feature request since officially opus files _are_ supported].

**Comment by Aga-C** [comment]
> The files can be played on VLC so I don't think it's an OS issue.

@firubat Can you play them in other music players? Because VLC is a bad example - it's able to decode any file format without relying on device/OS support.

**Comment by Aga-C** [comment]
@firubat Please create a new bug report. If you can, attach to it the non-working OPUS file.

**Comment by firubat** [comment]
@Aga-C  after some more digging seems that the problem is not with opus files in general, but specifically with Whatsapp's PTT voice notes.

Opened an issue here https://github.com/FossifyOrg/Music-Player/issues/111
